applications/GeoMechanicsApplication/tests/cpp_tests/test_time_loop_executor.cpp

```diff
@@ -0,0 +1,356 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Wijtze Pieter Kikstra
+//                   Anne van de Graaf
+//
+
+#include "testing/testing.h"
+#include "custom_workflows/time_loop_executor.hpp"
+#include "custom_workflows/time_step_end_state.hpp"
+#include "custom_workflows/prescribed_time_incrementor.h"
+#include "containers/model.h"
+
+#include <numeric>
+
+using namespace Kratos;
+
+namespace
+{
+
+class DummySolverStrategy : public StrategyWrapper
+{
+public:
+    explicit DummySolverStrategy(TimeStepEndState::ConvergenceState AConvergenceState)
+    : mConvergenceState{AConvergenceState}
+    {
+        mModel.Reset();
+        mpModelPart = &mModel.CreateModelPart("TestModelPart");
+        mpModelPart->Reset();
+    }
+
+    [[nodiscard]] TimeStepEndState::ConvergenceState GetConvergenceState() override
+    {
+        return mConvergenceState;
+    }
+
+    [[nodiscard]] std::size_t GetNumberOfIterations() const override
+    {
+        return 1;
+    }
+
+    [[nodiscard]] double GetEndTime() const override
+    {
+        return mpModelPart->GetProcessInfo()[TIME];
+    }
+
+    void SetEndTime(double EndTime) override
+    {
+        mpModelPart->GetProcessInfo()[TIME] = EndTime;
+    }
+
+    [[nodiscard]] double GetTimeIncrement() const override
+    {
+        return mpModelPart->GetProcessInfo()[DELTA_TIME];
+    }
+
+    void SetTimeIncrement(double TimeIncrement) override
+    {
+        mpModelPart->GetProcessInfo()[DELTA_TIME] = TimeIncrement;
+    }
+
+    [[nodiscard]] std::size_t GetStepNumber() const override
+    {
+        return static_cast<std::size_t>(mpModelPart->GetProcessInfo()[STEP]);
+    }
+
+    void IncrementStepNumber() override
+    {
+        mpModelPart->GetProcessInfo()[STEP] += 1;
```

Comment:
Blaming myself here. I think we'd better use the pre-increment operator.
```suggestion
        ++mpModelPart->GetProcessInfo()[STEP];
```

Reply:
Done